ui/app/src/components/PathNavigatorTree/PathNavigatorTree.tsx

```diff
@@ -241,8 +252,33 @@ export default function PathNavigatorTree(props: PathNavigatorTreeProps) {
       switch (type) {
         case itemsPasted.type:
         case folderCreated.type: {
-          if (payload.clipboard.type === 'CUT') {
-            debugger;
+          if (payload.clipboard?.type === 'CUT') {
+            const sourceNode =
+              nodesByPathRef.current[getParentPath(payload.clipboard.sourcePath)] ??
+              nodesByPathRef.current[withIndex(getParentPath(payload.clipboard.sourcePath))];
+            const targetNode =
+              nodesByPathRef.current[payload.target] ?? nodesByPathRef.current[withIndex(payload.target)];
```

Comment:
Should we be doing rather this:
```suggestion
              nodesByPathRef.current[withoutIndex(payload.target)] ?? nodesByPathRef.current[withIndex(payload.target)];
```
We don't know if an items is registered with or without index on the state so if not using both `withoutIndex` and `withIndex` we might be checking for the same thing and not getting the result we want.

We should make an util that does this
```ts
export function lookupItemByPath(path: string, lookupTable: LookupTable<Sandbox>): Sandbox
export function lookupItemByPath(path: string, lookupTable: LookupTable<Detailed>): Detailed 
export function lookupItemByPath(path: string, lookupTable: LookupTable<Sandbox | Detailed>): Sandbox | Detailed {
  return lookupTable[withoutIndex(path)] ?? lookupTable[withInde(path)];
}
```

Same deal would apply above to `sourceNode`. See if there's anywhere else that could use the new util.